### PR TITLE
Clang tidy fixes

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

File: couchbase/analytics_index_manager.hxx [hunk 1]
```diff
@@ -91,7 +91,7 @@ public:
    * @since 1.0.0
    * @committed
    */
-  void drop_dataverse(std::string dataverse_name,
+  void drop_dataverse(const std::string& dataverse_name,
```
Comment (DemetrisChr) [inline couchbase/analytics_index_manager.hxx:94]: Is turning `std::string` to `const std::string&` a breaking change?

Comment (avsej) [reply couchbase/analytics_index_manager.hxx:94]: Technically it is, as the compiled code will not be able to find the signature as it is using different type. But I think I'd rather make it early and right. Right now we do not ship binaries anyway yet. And the user would need to recompile it anyway.

File: couchbase/query_index_manager.hxx [hunk 2]
```diff
@@ -88,7 +88,7 @@ public:
    * @since 1.0.0
    * @committed
    */
-  void create_index(std::string bucket_name,
+  void create_index(const std::string& bucket_name,
                     std::string index_name,
                     std::vector<std::string> keys,
```
Comment (DemetrisChr) [inline couchbase/query_index_manager.hxx:93]: Should the rest of the parameters here and in the methods below also become const references?

File: core/transactions/transactions_config.cxx [hunk 3]
```diff
@@ -25,8 +25,7 @@ namespace couchbase::transactions
 {
 
 transactions_config::transactions_config()
-  : level_(couchbase::durability_level::majority)
-  , timeout_(std::chrono::seconds(15))
+  : timeout_(std::chrono::seconds(15))
```
Comment (DemetrisChr) [inline core/transactions/transactions_config.cxx:28]: Should this also be removed, similar to `level_`? It's being initialized to 15 seconds by default.

```c++
  std::chrono::nanoseconds timeout_{ std::chrono::seconds(15) };
```

File: couchbase/query_index_manager.hxx (hunk 2 above)
Comment (avsej) [reply couchbase/query_index_manager.hxx:93]: Probably they should. While taking by value and moving it later looks more efficient, we still transform those keys into JSON at some point, and treat them as temporary values anyway, so it is kind of waste of the memory. 

So in this context, taking it by reference would be more reasonable, as we do not want to actually use this chunk of memory for a long time (as a part of request), but rather convert it into some internal representation in some front-end.

File: core/impl/fail_fast_retry_strategy.cxx [hunk 4]
```diff
@@ -15,9 +15,15 @@
  *   limitations under the License.
  */
 
+#include "couchbase/retry_action.hxx"
+#include "couchbase/retry_reason.hxx"
+#include "couchbase/retry_request.hxx"
```
Comment (DemetrisChr) [inline core/impl/fail_fast_retry_strategy.cxx:20]: Should `<>` be used for these includes?

Comment (avsej) [reply core/impl/fail_fast_retry_strategy.cxx:20]: yes, in general I try to follow it as a style rule with <> for our "public API" and double quotes for the other parts, thanks for spotting it.

**Comment by avsej** [comment]
jenkins, test this please

**Pull request merged**